### PR TITLE
invoke animation listener when camera is already in or transitioning to requested state

### LINKED ISSUE #4341: AnimatorListener should be called when camera transition is already in progress
When `NavigationCamera#request[state]` methods are called and we're already transitioning, the method just returns and leaves the potentially provided `AnimatorListener` without a callback. Example:
https://github.com/mapbox/mapbox-navigation-android/blob/354689d14d6fb08b5c562e87a5849af3257e31c9/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/NavigationCamera.kt#L167-L168

We should consider calling some method of the callback or add it to the list of callbacks to invoke when the transition finishes.

### PR DESCRIPTION
**Pull request by Zayankovsky**
### Description

Fixes #4341. 

### Changelog

```
<changelog>Replaced `AnimatorListener` argument of `NavigationCamera#request[state]` functions with `TransitionEndListener`, which is now invoked even if the camera is already in or transitioning to the requested state</changelog>
```

### CODE REVIEW

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/NavigationCamera.kt [hunk 1]
```diff
@@ -122,7 +122,12 @@ class NavigationCamera(
             NavigationCameraTransitionOptions.Builder().maxDuration(1000L).build()
     }
 
+    fun interface TransitionEndListener {
```
Comment (RingerJK) [inline libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/NavigationCamera.kt:125]: let's move to separate file `TransitionEndListener ` and add docs

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/NavigationCamera.kt [hunk 2]
```diff
@@ -198,18 +201,23 @@ class NavigationCamera(
...
     ) {
         when (state) {
-            TRANSITION_TO_FOLLOWING, FOLLOWING -> {
-                // this should invoke animatorListener
-                // https://github.com/mapbox/mapbox-navigation-android/issues/4341
+            TRANSITION_TO_FOLLOWING -> {
+                if (transitionEndListener != null) {
+                    runningTransitionEndListeners.add(transitionEndListener)
+                }
+                return
```
Comment (RingerJK) [inline libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/NavigationCamera.kt:217]: unnecessary `return` here and below

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/NavigationCamera.kt [hunk 3]
```diff
@@ -273,18 +279,23 @@ class NavigationCamera(
...
     ) {
         when (state) {
-            TRANSITION_TO_OVERVIEW, OVERVIEW -> {
-                // this should invoke animatorListener
-                // https://github.com/mapbox/mapbox-navigation-android/issues/4341
+            TRANSITION_TO_OVERVIEW -> {
+                if (transitionEndListener != null) {
+                    runningTransitionEndListeners.add(transitionEndListener)
+                }
+                return
```
Comment (RingerJK) [inline libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/NavigationCamera.kt:295]: unnecessary `return` here and below

File: libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/NavigationCamera.kt [hunk 4]
```diff
@@ -422,15 +433,14 @@ class NavigationCamera(
         progressState: NavigationCameraState,
         finalState: NavigationCameraState,
         frameTransitionOptions: NavigationCameraTransitionOptions,
-        externalListener: Animator.AnimatorListener?
+        transitionEndListener: TransitionEndListener?,
```
Comment (RingerJK) [inline libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/NavigationCamera.kt:436]: avoid set implicitly `runningTransitionEndListeners` and add any of the listeners to `runningTransitionEndListeners` that have already been used

**Review by LukasPaczos — APPROVED** [review]
LGTM :+1:  let's add a changelog entry as this is a breaking change.

**Review by RingerJK — APPROVED** [review]

**Comment by Zayankovsky** [comment]
> LGTM 👍 let's add a changelog entry as this is a breaking change.

Added.

**Pull request merged**